Martinez_Oscar_scraping1.ipynb

```diff
@@ -0,0 +1,181 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {
+    "autoscroll": false,
+    "ein.hycell": false,
+    "ein.tags": "worksheet-0",
+    "slideshow": {
+     "slide_type": "-"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# Our jupyter/datascience-notebook Docker container comes with \n",
+    "# BeautifulSoup4 and requests, both popular libraries!\n",
+    "\n",
+    "from bs4 import BeautifulSoup\n",
+    "import requests as rq\n",
+    "import re"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {
+    "autoscroll": false,
+    "ein.hycell": false,
+    "ein.tags": "worksheet-0",
+    "slideshow": {
+     "slide_type": "-"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "START_URL = 'https://brickset.com/sets/year-2016'"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 90,
+   "metadata": {
+    "autoscroll": false,
+    "ein.hycell": false,
+    "ein.tags": "worksheet-0",
+    "slideshow": {
+     "slide_type": "-"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "def get_titles(soup): #this first function is going to find all the objects with class meta and type div  \n",
+    "    class_meta=soup.find_all(\"div\",{'class':'meta'}) #which corresponds to the class where the titles are\n",
+    "    titles=[]\n",
+    "    for i in class_meta: #for each element e stract the h1 (which is a header)\n",
+    "        titles.append(i.find(\"h1\").text)\n",
+    "    return titles\n",
+    "\n",
+    "def parse_bricks(url): #this function takes a url, takes the content and transforms it to a soup object\n",
+    "    \"\"\" Fetches Lego Bricks page and extracts titles and next page url \"\"\"\n",
+    "    c=rq.get(url) \n",
+    "    result=c.content\n",
+    "    soup=BeautifulSoup(result)\n",
+    "    titles=get_titles(soup) #we put the soup object into our function to obtain the titles\n",
+    "    \n",
+    "    class_page=soup.find_all(\"li\",{'class':'next'}) #we find all the url from the next class (next page)\n",
+    "    try:\n",
+    "        next_url=class_page[0].find(\"a\").get(\"href\")#we have two next buttons so we stick with the first one and get the url\n",
+    "    except:\n",
+    "        next_url=0\n",
+    "    return (titles,next_url)"
```

Comment:
Great returning of a tuple, very Pythonic! No need for parenthesis, but that's stylistic.